### PR TITLE
Add new function shell_tools.run

### PR DESCRIPTION
**Pull request by pavoljuhas**
Add shell_tools.run, a thin wrapper to subprocess.run with
a few extra arguments for showing the executed commands.
It also supplies updated default values for subprocess.run
to approximate behavior of run_cmd and run_shell functions.
The purpose of this function is to replace the run_cmd and
run_shell functions in the shell_tools module.

Partially implements https://github.com/quantumlib/Cirq/issues/4394

### CODE REVIEW

**Comment by pavoljuhas** [comment]
Premature PR - please hold with review...

File: dev_tools/shell_tools.py [hunk 1]
```diff
@@ -118,6 +118,53 @@ def abbreviate_command_arguments_after_switches(cmd: Tuple[str, ...]) -> Tuple[s
...
+    args: Union[str, List[str]],
+    *,
+    log_run_to_stderr: bool = True,
+    abbreviate_non_option_arguments: bool = False,
+    check: bool = True,
+    text: bool = True,
+    **subprocess_run_kwargs,
+) -> subprocess.CompletedProcess:
+    """Call subprocess.run with an option to log executed command to stderr.
+
```
Comment (dabacon) [inline dev_tools/shell_tools.py:131]: maybe something to distinguish why you would use this and not run_cmd below?

**Review by dabacon — APPROVED** [review]
LGTM

One thought on documentation, but however you decide is good.

File: dev_tools/shell_tools.py (hunk 1 above)
Comment (pavoljuhas) [reply dev_tools/shell_tools.py:131]: Ack.  I actually plan to remove run_cmd and run_shell in a followup PR and only use the `run()` function.
This PR is 1 of 4 for [#4394](https://github.com/quantumlib/Cirq/issues/4394#issuecomment-1155868381).

Thanks for quick review.

**Pull request merged**